<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Profile likelihood â€¢ pulseR</title>
<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Profile likelihood">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pulseR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workflow.html">The workflow</a>
    </li>
    <li>
      <a href="../articles/epitranscriptomics.html">Analysis of RNA modifications</a>
    </li>
    <li>
      <a href="../articles/confidence.html">Profile likelihood</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dieterich-lab/pulseR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Profile likelihood</h1>
                        <h4 class="author">Uvarovskii Alexey</h4>
            
            <h4 class="date">2018-10-22</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dieterich-lab/pulseR/blob/master/vignettes/confidence.Rmd"><code>vignettes/confidence.Rmd</code></a></small>
      <div class="hidden name"><code>confidence.Rmd</code></div>

    </div>

    
    
<div id="approximation-of-confidence-intervals-using-only-gene-specific-data" class="section level2">
<h2 class="hasAnchor">
<a href="#approximation-of-confidence-intervals-using-only-gene-specific-data" class="anchor"></a>Approximation of confidence intervals using only gene specific data</h2>
<div id="what-does-it-mean-and-when-can-i-use-it" class="section level3">
<h3 class="hasAnchor">
<a href="#what-does-it-mean-and-when-can-i-use-it" class="anchor"></a>What does it mean and when can I use it?</h3>
<p>The profile likelihood is estimated by optimising the whole likelihood function, having fixed a parameter under consideration to a certain value. Parameter values for different genes influence each other via shared parameters, for example, the normalisation factors or variance of the negative binomial distribution.</p>
<p>If there are not many genes in the data, that could not be a prolblem. However, usually experimentalists doing RNA-seq experiments are interested in measuring parameters of as many genes as possible, which results in thousands of genes in data tables. Optimising likelihood function for such cases can take a lot of computation time. Hence it can be more efficeint to use approximate fitting by neglecting the influence of individual gene parameters on the shared parameters.</p>
</div>
<div id="the-workflow" class="section level3">
<h3 class="hasAnchor">
<a href="#the-workflow" class="anchor"></a>The workflow</h3>
<p>Let us use the model of experiment used in the <a href="workflow.html">vignette</a> for application of pulseR to spike-ins-free data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(pulseR)
<span class="kw">attach</span>(pulseRSpikeinsData)

formulas &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MeanFormulas.html">MeanFormulas</a></span>(
  <span class="dt">total      =</span>  mu,
  <span class="dt">labelled   =</span>  mu *<span class="st"> </span><span class="kw">exp</span>(-d *<span class="st"> </span>time),
  <span class="dt">unlabelled =</span>  mu *<span class="st"> </span>(<span class="dv">1</span> -<span class="st"> </span><span class="kw">exp</span>(-d *<span class="st"> </span>time))
)
formulaIndexes &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">total_fraction =</span> <span class="st">"total"</span>,
                       <span class="dt">pull_down      =</span> <span class="kw">c</span>(<span class="st">"labelled"</span>, <span class="st">"unlabelled"</span>))
pd &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PulseData.html">PulseData</a></span>(counts, conditions, formulas, formulaIndexes, 
                <span class="dt">spikeins =</span> spikeins)
opts &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setBoundaries.html">setBoundaries</a></span>(<span class="kw">list</span>(
  <span class="dt">mu =</span> <span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">1e6</span>),
  <span class="dt">d =</span> <span class="kw">c</span>(.<span class="dv">01</span>, <span class="dv">2</span>)
))
opts$ub$size &lt;-<span class="st"> </span><span class="fl">1e7</span>
opts &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setTolerance.html">setTolerance</a></span>(<span class="dt">params =</span> <span class="fl">1e-3</span>,  <span class="dt">options =</span> opts)
initPars &lt;-<span class="st"> </span><span class="kw"><a href="../reference/initParameters.html">initParameters</a></span>(<span class="dt">par =</span> <span class="kw">list</span>(<span class="dt">size=</span><span class="dv">100</span>),
                           <span class="dt">geneParams =</span> <span class="kw">c</span>(<span class="st">"mu"</span>, <span class="st">"d"</span>),
                           <span class="dt">pulseData  =</span> pd,
                           <span class="dt">options    =</span> opts)
initPars$d[] &lt;-<span class="st"> </span>.<span class="dv">1</span>
initPars$mu &lt;-<span class="st"> </span>pd$counts[,<span class="dv">1</span>] +<span class="st"> </span><span class="dv">1</span>
fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/fit.html">fitModel</a></span>(<span class="dt">pulseData =</span> pd, <span class="dt">par =</span> initPars, <span class="dt">options =</span> opts)</code></pre></div>
<div id="construct-the-profile-likelihood" class="section level4">
<h4 class="hasAnchor">
<a href="#construct-the-profile-likelihood" class="anchor"></a>Construct the profile likelihood</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">geneIndex &lt;-<span class="st"> </span><span class="dv">7</span>
parName &lt;-<span class="st"> "d"</span>
interval &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.8</span>,<span class="fl">1.3</span>) *<span class="st"> </span>fit[[parName]][geneIndex]
numPoints &lt;-<span class="st"> </span><span class="dv">51</span>
pl &lt;-<span class="st"> </span><span class="kw"><a href="../reference/profile.html">profileGene</a></span>(parName, geneIndex, pd, fit, interval, 
                  <span class="dt">options=</span>opts, <span class="dt">numPoints =</span> <span class="dv">21</span>)
<span class="kw">str</span>(pl)
## 'data.frame':    21 obs. of  2 variables:
##  $ d   : num  0.124 0.127 0.131 0.135 0.139 ...
##  $ logL: num  19.23 14.48 10.41 7.01 4.29 ...</code></pre></div>
<p>We can see the result using the <code>plotPL</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plotPL.html">plotPL</a></span>(pl)</code></pre></div>
<p><img src="confidence_files/figure-html/unnamed-chunk-4-1.png" width="480"></p>
<p>The red line corresponds to the 95% confidence level likelihood threshold. The 95%-confidence interval for the <span class="math inline">\(d\)</span> parameter is defined from the crossing points of the threshold line and the profile likelihood curve.</p>
<p>Use the <code>ciGene</code> function in order to estimate this points numerically:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">geneIndexes &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="dv">10</span>
CIs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/confint.html">ciGene</a></span>(parName, geneIndexes, pd, fit, opts)
CIs
##            [,1]      [,2]
##  [1,]        NA        NA
##  [2,]        NA        NA
##  [3,] 0.2342371 0.2437456
##  [4,] 0.1083609 0.1290361
##  [5,] 0.1968716 0.2190968
##  [6,] 0.1536246 0.1719114
##  [7,] 0.1435240 0.1619218
##  [8,] 0.2693011 0.2887113
##  [9,] 0.2757496 0.2946228
## [10,] 0.2386713 0.2568335</code></pre></div>
</div>
</div>
</div>
<div id="estimation-of-confidence-intervals-using-the-whole-likelihood" class="section level2">
<h2 class="hasAnchor">
<a href="#estimation-of-confidence-intervals-using-the-whole-likelihood" class="anchor"></a>Estimation of confidence intervals using the whole likelihood</h2>
<p>If the normalisation factors are not fixed, or changes in a single gene parameter result in observed difference in the shared or normalisation parameters values, it is necessary to use the whole likelihood, which is a function of all shared and individual parameters.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">plTotal &lt;-<span class="st"> </span><span class="kw"><a href="../reference/profile.html">profile</a></span>(<span class="dt">paramPath =</span> <span class="kw">list</span>(<span class="st">"d"</span>,geneIndex),
              <span class="dt">pd =</span> pd, 
              <span class="dt">par =</span> fit,
              <span class="dt">options =</span> opts,
              <span class="dt">interval =</span> interval,
              <span class="dt">numPoints =</span> <span class="dv">20</span>)</code></pre></div>
<p>If the model includes the normalisation factors and we are interested in estimation of the CI for the normalisation factors, it is possible to run the same function in the same manner:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">plTotal &lt;-<span class="st"> </span><span class="kw"><a href="../reference/profile.html">profile</a></span>(<span class="dt">paramPath =</span> <span class="kw">list</span>(<span class="st">"normFactors"</span>,<span class="dv">2</span>,<span class="dv">1</span>),
              <span class="dt">pd =</span> pd, 
              <span class="dt">par =</span> fit,
              <span class="dt">options =</span> opts,
              <span class="dt">interval =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</code></pre></div>
<p>However, it may result in long computation times, since it is equivalent to refitting the whole model several times.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
## R version 3.4.0 (2017-04-21)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.5 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/openblas-base/libblas.so.3
## LAPACK: /usr/lib/libopenblasp-r0.2.18.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] pulseR_1.0.3
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.19       roxygen2_6.1.0     crayon_1.3.4      
##  [4] assertthat_0.2.0   digest_0.6.18      rprojroot_1.3-2   
##  [7] commonmark_1.6     MASS_7.3-50        R6_2.3.0          
## [10] backports_1.1.2    magrittr_1.5       evaluate_0.12     
## [13] stringi_1.2.4      rlang_0.2.2        fs_1.2.6          
## [16] xml2_1.2.0         rmarkdown_1.10     pkgdown_1.1.0.9000
## [19] desc_1.2.0         tools_3.4.0        stringr_1.3.1     
## [22] yaml_2.2.0         compiler_3.4.0     memoise_1.1.0     
## [25] htmltools_0.3.6    knitr_1.20</code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#approximation-of-confidence-intervals-using-only-gene-specific-data">Approximation of confidence intervals using only gene specific data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#what-does-it-mean-and-when-can-i-use-it">What does it mean and when can I use it?</a></li>
      <li>
<a href="#the-workflow">The workflow</a><ul class="nav nav-pills nav-stacked">
<li><a href="#construct-the-profile-likelihood">Construct the profile likelihood</a></li>
      </ul>
</li>
      </ul>
</li>
      <li><a href="#estimation-of-confidence-intervals-using-the-whole-likelihood">Estimation of confidence intervals using the whole likelihood</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Alexey Uvarovskii.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
