<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Application to epitranscriptomics • pulseR</title>
<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Application to epitranscriptomics">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pulseR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workflow.html">The workflow</a>
    </li>
    <li>
      <a href="../articles/epitranscriptomics.html">Analysis of RNA modifications</a>
    </li>
    <li>
      <a href="../articles/confidence.html">Profile likelihood</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dieterich-lab/pulseR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Application to epitranscriptomics</h1>
                        <h4 class="author">Uvarovskii Alexey</h4>
            
            <h4 class="date">17-10-2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dieterich-lab/pulseR/blob/master/vignettes/epitranscriptomics.Rmd"><code>vignettes/epitranscriptomics.Rmd</code></a></small>
      <div class="hidden name"><code>epitranscriptomics.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The <code>pulseR</code> package may be adapted for analysis of experimental approaches, which are similar to the metabolic labelling in having several related fractions. For example, RNA modification level can be measured on the basis of immunoprecipitation (IP) using antibodies against of modified nucleotisides, e.g. anti-m<span class="math inline">\(^6\)</span>A or anti-m<span class="math inline">\(^5\)</span>C IgG.</p>
<p>Here we consider an example of the LAIC-seq approach, <a href="https://www.nature.com/articles/nmeth.3898">published here</a></p>
<blockquote>
<p>Molinie, Benoit, et al. “m(6)A-LAIC-seq reveals the census and complexity of the m(6)A epitranscriptome.” Nature methods 13.8 (2016): 692.</p>
</blockquote>
<p>The data consists of three fractions:</p>
<ul>
<li>total (“input”) <span class="math inline">\(\leftarrow\)</span> the original pool</li>
<li>pull-down (“elution”) <span class="math inline">\(\leftarrow\)</span> m<span class="math inline">\(^6\)</span>A modified RNA</li>
<li>flow-through (“supernatant”) <span class="math inline">\(\leftarrow\)</span> m<span class="math inline">\(^6\)</span>A <strong>non</strong>modified RNA.</li>
</ul>
<p>Since during library preparation, usually the probes are normalised to the same amount (e.g. 100ng), it is important to recover the original relation between the pull-down, flow-through and the total fractions. One may whether</p>
<ul>
<li>use external molecules for normalisation (e.g. ERCC spike-ins or RNA from another species). The approaches are
<ol style="list-style-type: decimal">
<li>external spike-in molecules are added in known (e.g. equal) proportion <strong>right after the purification</strong> (<span class="math inline">\(\leftarrow\)</span> approach in the original paper). One must prove experimentally, that efficiency of separation is high (negligible cross-contamination)</li>
<li>nonmodified and modified (e.g. m<span class="math inline">\(^6\)</span>A<span class="math inline">\(^+\)</span> RNA) spike-ins are <strong>before the purification</strong>
</li>
</ol>
</li>
<li>estimate relations between fractions during model fitting ( <span class="math inline">\(\leftarrow\)</span> this example)</li>
</ul>
<p>The task of normalisation and the model fitting can be illustrated with the following figure:</p>
<p><img src="fig/laicseq-model.png" style="width: 500px"></p>
<p>Considering the total sample as a reference, with denote the expression level of a given gene as <span class="math inline">\(\mu\)</span>. Then if <span class="math inline">\(\alpha\)</span> is proportion of modified molecules (a value between 0 and 1), the amount of the modified molecules in the total sample is <span class="math inline">\(\alpha\mu\)</span> and amount of the nonmodified molecules is <span class="math inline">\((1 - \alpha)\mu\)</span>.</p>
<p>However, normalisation of material for library preparation affects the expected read counts in both fractions, and the task is to recover additional normalisation factors <span class="math inline">\(x_1\)</span> and <span class="math inline">\(x_2\)</span> (see figure), in such a way, that the read counts can be explained by the model. The coefficients <span class="math inline">\(x_1\)</span> and <span class="math inline">\(x_2\)</span> are shared between all the genes, and the expression level and modification level are gene-specific.</p>
</div>
<div id="analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis" class="anchor"></a>Analysis</h2>
<div id="get-and-prepare-the-data" class="section level3">
<h3 class="hasAnchor">
<a href="#get-and-prepare-the-data" class="anchor"></a>Get and prepare the data</h3>
<p>Thanks to the authors, the paper is accompanied by the supplementary count table, which makes reproducibility much easier. In this subsection, preliminary steps for data preparation are discussed. Download the table with read counts and results from the publisher’s site:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(openxlsx)
dataURL &lt;-<span class="st"> "https://media.nature.com/original/nature-assets/nmeth/journal/v13/n8/extref/nmeth.3898-S2.xlsx"</span>
laicSeqData &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/openxlsx/topics/read.xlsx">read.xlsx</a></span>(dataURL)</code></pre></div>
<p>Prepare the condition table and the count table:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts &lt;-<span class="st"> </span>laicSeqData[, -(<span class="dv">1</span>:<span class="dv">10</span>)]
<span class="kw">rownames</span>(counts) &lt;-<span class="st"> </span>laicSeqData$Gene_ID

conditions &lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind, <span class="kw">strsplit</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/openxlsx/topics/names">names</a></span>(counts), <span class="st">"_"</span>))
conditions &lt;-<span class="st"> </span><span class="kw">data.frame</span>(conditions[,<span class="dv">3</span>:<span class="dv">1</span>], <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/openxlsx/topics/names">names</a></span>(conditions) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"fraction"</span>, <span class="st">"run"</span>, <span class="st">"cells"</span>)
conditions$sample &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/openxlsx/topics/names">names</a></span>(counts)

conditions$fraction &lt;-<span class="st"> </span><span class="kw">c</span>(
 <span class="st">"input"</span> =<span class="st"> "total"</span>,
 <span class="st">"elu"</span>   =<span class="st"> "pull-down"</span>,
 <span class="st">"sup"</span>   =<span class="st"> "flow-through"</span>)[conditions$fraction]

## In this example we use only the "GM" cells data set
cellTypeIndex &lt;-<span class="st"> </span>conditions$cells ==<span class="st"> "GM"</span>
byFraction &lt;-<span class="st"> </span><span class="kw">order</span>(conditions$fraction[cellTypeIndex], <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)

counts     &lt;-<span class="st"> </span>counts[, cellTypeIndex][,byFraction]
conditions &lt;-<span class="st"> </span>conditions[cellTypeIndex,][byFraction,]

knitr::<span class="kw">kable</span>(conditions, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">fraction</th>
<th align="left">run</th>
<th align="left">cells</th>
<th align="left">sample</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">total</td>
<td align="left">sp1</td>
<td align="left">GM</td>
<td align="left">GM_sp1_input_pairs</td>
</tr>
<tr class="even">
<td align="left">total</td>
<td align="left">sp2</td>
<td align="left">GM</td>
<td align="left">GM_sp2_input_pairs</td>
</tr>
<tr class="odd">
<td align="left">pull-down</td>
<td align="left">sp1</td>
<td align="left">GM</td>
<td align="left">GM_sp1_elu_pairs</td>
</tr>
<tr class="even">
<td align="left">pull-down</td>
<td align="left">sp2</td>
<td align="left">GM</td>
<td align="left">GM_sp2_elu_pairs</td>
</tr>
<tr class="odd">
<td align="left">flow-through</td>
<td align="left">sp1</td>
<td align="left">GM</td>
<td align="left">GM_sp1_sup_pairs</td>
</tr>
<tr class="even">
<td align="left">flow-through</td>
<td align="left">sp2</td>
<td align="left">GM</td>
<td align="left">GM_sp2_sup_pairs</td>
</tr>
</tbody>
</table>
<p>The count table looks like</p>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">GM_sp1_input_pairs</th>
<th align="right">GM_sp2_input_pairs</th>
<th align="right">GM_sp1_elu_pairs</th>
<th align="right">GM_sp2_elu_pairs</th>
<th align="right">GM_sp1_sup_pairs</th>
<th align="right">GM_sp2_sup_pairs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000167634</td>
<td align="right">103</td>
<td align="right">140</td>
<td align="right">43</td>
<td align="right">42</td>
<td align="right">143</td>
<td align="right">201</td>
</tr>
<tr class="even">
<td align="left">ENSG00000104951</td>
<td align="right">2951</td>
<td align="right">3823</td>
<td align="right">5929</td>
<td align="right">5244</td>
<td align="right">2521</td>
<td align="right">4282</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000116096</td>
<td align="right">112</td>
<td align="right">130</td>
<td align="right">153</td>
<td align="right">172</td>
<td align="right">87</td>
<td align="right">150</td>
</tr>
<tr class="even">
<td align="left">ENSG00000137496</td>
<td align="right">741</td>
<td align="right">787</td>
<td align="right">1350</td>
<td align="right">1145</td>
<td align="right">476</td>
<td align="right">578</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000123933</td>
<td align="right">398</td>
<td align="right">536</td>
<td align="right">281</td>
<td align="right">263</td>
<td align="right">456</td>
<td align="right">681</td>
</tr>
<tr class="even">
<td align="left">ENSG00000266728</td>
<td align="right">747</td>
<td align="right">810</td>
<td align="right">738</td>
<td align="right">597</td>
<td align="right">756</td>
<td align="right">724</td>
</tr>
</tbody>
</table>
</div>
<div id="which-parametrisation-to-use" class="section level3">
<h3 class="hasAnchor">
<a href="#which-parametrisation-to-use" class="anchor"></a>Which parametrisation to use?</h3>
<p>There are several ways to report the methylation level:</p>
<ul>
<li>as a <strong>proportion</strong> <span class="math inline">\(\alpha\)</span> of the molecules, which are trapped by the antibodies.<br>
In this case, <span class="math inline">\(\alpha\)</span> is in [0,1] interval.</li>
<li>as a <strong>ratio</strong> of modified / nonmodified molecules.<br>
It is equal to <span class="math inline">\(\alpha/(1 - \alpha)\)</span> and is in <span class="math inline">\([0,\infty)\)</span> interval</li>
<li>as the <strong>logarithm of odds</strong>, <span class="math inline">\(\gamma = \log\left(\dfrac{\alpha}{1 - \alpha}\right)\)</span> <span class="math inline">\(\leftarrow\)</span> <strong>we are here</strong>.<br>
It can take all the values from <span class="math inline">\((-\infty, \infty)\)</span>.</li>
</ul>
<p>Out of these three parameters, only the log(odds) is symmetrical in respect to the modified and unmodified molecules quantities. If, for example, the level of modification increases from 80% to 90%, it is only a mild change in the proportion (1.125-fold), but the amount of nonmodified molecules decreased from 20% to 10%, which is 2-fold decrease. Hence, the extent of the difference depends on the molecule type, we are considering.</p>
<p>In contrast, if we swap the labels of the molecules in the log(odds), it results only in change of the sign, but not the absolute value:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lor &lt;-<span class="st"> </span>function(a) <span class="kw">log</span>(a/(<span class="dv">1</span> -<span class="st"> </span>a))
<span class="kw">lor</span>(.<span class="dv">9</span>) -<span class="st"> </span><span class="kw">lor</span>(.<span class="dv">8</span>) <span class="co"># change from 80% to 90%</span>
<span class="co">#&gt; [1] 0.8109302</span>
<span class="kw">lor</span>(.<span class="dv">1</span>) -<span class="st"> </span><span class="kw">lor</span>(.<span class="dv">2</span>) <span class="co"># change from 20% to 10%</span>
<span class="co">#&gt; [1] -0.8109302</span></code></pre></div>
<p>It is always good to keep gene-specific parameters at the same scale, since currently the stopping criteria for fitting depends on the absolute changes in parameter values between iterations. That it is why, if the log(odds) is used, it is better to describe the expression level at the logarithmic scale, i.e. mean read count is <span class="math display">\[\text{[total]} = e^{\mu}\]</span> and the amount of modified molecules is <span class="math display">\[\text{[modified]} = e^{\mu + \gamma} / (1 + e^{\gamma}).\]</span></p>
</div>
<div id="model-definition" class="section level3">
<h3 class="hasAnchor">
<a href="#model-definition" class="anchor"></a>Model definition</h3>
<p><strong>In this workflow, we use the parametrisation based on the log(odds).</strong> To have parameters of similar scale, the expression level is modelled at the logarithmic scale, i.e. mean read count in the total fraction is <code>exp(mu)</code>.</p>
<p>To use the <code>pulseR</code> package, we need to define these formulas and describe, how the read counts in different conditions are related to the formulas:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(pulseR)
formulas &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MeanFormulas.html">MeanFormulas</a></span>(
  <span class="dt">total       =</span> <span class="kw">exp</span>(mu),
  <span class="dt">modified    =</span> <span class="kw">exp</span>(y +<span class="st"> </span>mu) /<span class="st"> </span>(<span class="dv">1</span> +<span class="st"> </span><span class="kw">exp</span>(y)),
  <span class="dt">nonmodified =</span> <span class="kw">exp</span>(mu) /<span class="st"> </span>(<span class="dv">1</span> +<span class="st"> </span><span class="kw">exp</span>(y))
)
formIndexes &lt;-<span class="st"> </span><span class="kw">list</span>(
  <span class="st">"total"</span>        =<span class="st"> "total"</span>,
  <span class="st">"pull-down"</span>    =<span class="st"> "modified"</span>,
  <span class="st">"flow-through"</span> =<span class="st"> "nonmodified"</span>
)</code></pre></div>
<p>It is important to note, that here we neglect cross-contamination, as in the paper it was experimentally shown, that the purification procedure is very efficient, which results in high quality separation.</p>
<p>Alternatively, one may use spike-ins for estimation of the cross-contamination levels and relation of fractions to each other.</p>
<p>To keep computations light, we use only highly expressed genes, using mean read count in the total fraction for filtering:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">expressionRank &lt;-<span class="st"> </span><span class="kw">rank</span>(
  -<span class="kw">rowMeans</span>(counts[, conditions$fraction ==<span class="st"> "total"</span>]))
highExp &lt;-<span class="st"> </span>expressionRank &lt;<span class="st"> </span><span class="dv">501</span></code></pre></div>
<p>Now we prepared everything for the <code>PulseData</code> object initialisation:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pd &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PulseData.html">PulseData</a></span>(<span class="dt">counts =</span> counts[highExp,],
                <span class="dt">conditions =</span> conditions[, <span class="dv">1</span>,  <span class="dt">drop =</span> <span class="ot">FALSE</span>],
                <span class="dt">formulas =</span> formulas,
                <span class="dt">formulaIndexes =</span> formIndexes,
                <span class="dt">groups =</span> ~fraction)</code></pre></div>
</div>
<div id="fitting-options" class="section level3">
<h3 class="hasAnchor">
<a href="#fitting-options" class="anchor"></a>Fitting options</h3>
<p>The model is fitted using the optimisation routine “L-BFGS-B”, which requires boundaries for the parameters. In this case, there are following parameters to fit:</p>
<ul>
<li>gene specific: logarithmic expression level <span class="math inline">\(\mu\)</span> and the log(odds) <span class="math inline">\(\gamma\)</span> for RNA modification</li>
<li>size parameter for the negative binomial distribution (shared between all the genes)</li>
<li>normalisation factors (shared between the samples from the same condition, i.e. a single factor for e.g. “pull-down” samples). Alternatively, every sample can be attributed with its own normalisation factor, if, for example, efficiency of purification is varies between repetitions (batches).</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
lbNormFactors &lt;-<span class="st"> </span><span class="kw">list</span>(
  <span class="st">"total"</span>        =<span class="st"> </span>.<span class="dv">1</span>,
  <span class="st">"pull-down"</span>    =<span class="st"> </span>.<span class="dv">1</span>,
  <span class="st">"flow-through"</span> =<span class="st"> </span>.<span class="dv">1</span>
)
ubNormFactors &lt;-<span class="st"> </span><span class="kw">list</span>(
  <span class="st">"total"</span>        =<span class="st"> </span><span class="fl">2e1</span>,
  <span class="st">"pull-down"</span>    =<span class="st"> </span><span class="fl">2e1</span>,
  <span class="st">"flow-through"</span> =<span class="st"> </span><span class="fl">2e1</span>
)

opts &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setBoundaries.html">setBoundaries</a></span>(<span class="kw">list</span>(
  <span class="dt">mu =</span> <span class="kw">log</span>(<span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">1e8</span>)),
  <span class="dt">y =</span> <span class="kw">c</span>(-<span class="dv">5</span>,<span class="dv">5</span>),
  <span class="dt">size =</span> <span class="kw">c</span>(.<span class="dv">1</span>,<span class="fl">1e6</span>)),
  <span class="dt">normFactors =</span> <span class="kw">list</span>(lbNormFactors, ubNormFactors)
)
## Set stopping criteria for fitting:
## absolute difference between iterations must be not higher than 0.1 for
## the parameters (mu and y) and the normalisation factors
opts &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setTolerance.html">setTolerance</a></span>(<span class="dt">params =</span> .<span class="dv">1</span>, 
                     <span class="dt">options =</span> opts,
                     <span class="dt">normFactors =</span> .<span class="dv">1</span>,
                     <span class="dt">logLik =</span> <span class="dv">1</span> <span class="co"># minimal absolute difference for the logLikelihood</span>
)
opts$verbose &lt;-<span class="st"> "silent"</span>
opts$resultRDS &lt;-<span class="st"> </span><span class="ot">NULL</span> <span class="co"># a path to save the result after every iteration</span>
opts$cores &lt;-<span class="st"> </span><span class="dv">3</span></code></pre></div>
<p>To start model fitting, it can be beneficial to initialise the parameters close to their expected values: for example, the parameter <code>mu</code> must be near the logarithm of the read counts in the total fractions.</p>
<p>The normalisation factors will be fitted as well, except for the very first item: it is the reference for all other samples and will be fixed to 1 identifiability.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># create a list of needed structure and then adjust the values</span>
pars &lt;-<span class="st"> </span><span class="kw"><a href="../reference/initParameters.html">initParameters</a></span>(<span class="kw">list</span>(), <span class="kw">c</span>(<span class="st">"mu"</span>, <span class="st">"y"</span>), pd, opts)
pars$mu &lt;-<span class="st"> </span><span class="kw">log</span>(pd$counts[, <span class="dv">1</span>] +<span class="st"> </span><span class="dv">1</span>)
pars$y[] &lt;-<span class="st"> </span><span class="dv">0</span> <span class="co"># corresponds to 50% methylation</span>
pars$size &lt;-<span class="st"> </span><span class="dv">1000</span>
pars$normFactors[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="dv">2</span>
pars$normFactors[[<span class="dv">3</span>]] &lt;-<span class="st"> </span><span class="dv">2</span>

## look inside the initial guess
<span class="kw">str</span>(pars)
<span class="co">#&gt; List of 4</span>
<span class="co">#&gt;  $ mu         : Named num [1:500] 8.57 9.39 8.66 8.63 9.14 ...</span>
<span class="co">#&gt;   ..- attr(*, "names")= chr [1:500] "ENSG00000171992" "ENSG00000245532" "ENSG00000258486" "ENSG00000175265" ...</span>
<span class="co">#&gt;  $ y          : num [1:500] 0 0 0 0 0 0 0 0 0 0 ...</span>
<span class="co">#&gt;  $ size       : num 1000</span>
<span class="co">#&gt;  $ normFactors:List of 3</span>
<span class="co">#&gt;   ..$ : num 5.91</span>
<span class="co">#&gt;   ..$ : num 2</span>
<span class="co">#&gt;   ..$ : num 2</span></code></pre></div>
<p>Finally, run the fitting using the initial values, options and the data provided:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/fit.html">fitModel</a></span>(pd, pars, opts)</code></pre></div>
</div>
<div id="results" class="section level3">
<h3 class="hasAnchor">
<a href="#results" class="anchor"></a>Results</h3>
<p>We may compare how the model prediction and the raw counts relate to each other, using the <code>predictExpression</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/predictExpression.html">predictExpression</a></span>(result, pd)

<span class="kw">plot</span>(<span class="kw">as.vector</span>(pd$counts) +<span class="st"> </span>.<span class="dv">5</span>,
 <span class="kw">as.vector</span>(pr$predictions) +<span class="st"> </span>.<span class="dv">5</span>,
 <span class="dt">log=</span><span class="st">'xy'</span>,
 <span class="dt">cex =</span> .<span class="dv">3</span>,
 <span class="dt">xlab =</span> <span class="st">"experiment"</span>,
 <span class="dt">ylab =</span> <span class="st">"model"</span>)
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="dv">2</span>, <span class="dt">lwd =</span> <span class="dv">2</span>)</code></pre></div>
<p><img src="epitranscriptomics_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>In addition, in the paper supplementary the estimations for the methylation levels are available for two experiment repetitions. It is interesting to compare the pulseR estimations based on the both repetitions and the estimations from the paper.</p>
<p>Since we use only a subset of the data and a different way to normalise the fractions, it affects the estimations of the methylation level, hence a global bias between these two results can be observed, i.e. estimations will be shifted to higher or lower values. It complicates comparison of the results from, for example, different studies, and further development is needed in order to tackle this problem.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">paperResults &lt;-<span class="st"> </span>laicSeqData[highExp,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">6</span>,<span class="dv">7</span>)]
<span class="kw">head</span>(paperResults)
<span class="co">#&gt;     Gene_symbol GM_sp1_m6A_level GM_sp2_m6A_level</span>
<span class="co">#&gt; 12        SYNPO        0.4965471        0.5406506</span>
<span class="co">#&gt; 56        NEAT1        0.7849989        0.7952290</span>
<span class="co">#&gt; 60       RN7SL1        0.3111133        0.3156488</span>
<span class="co">#&gt; 122     GOLGA8A        0.5621528        0.5491124</span>
<span class="co">#&gt; 204       MACF1        0.5904389        0.5557059</span>
<span class="co">#&gt; 218        MAP4        0.6212910        0.6074293</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">or2alpha &lt;-<span class="st"> </span>function(or) <span class="kw">exp</span>(or) /<span class="st"> </span>(<span class="dv">1</span> +<span class="st"> </span><span class="kw">exp</span>(or))
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))
plotCompare &lt;-<span class="st"> </span>function(...) {
  <span class="kw">plot</span>(<span class="dt">cex =</span> .<span class="dv">5</span>, <span class="dt">pch =</span> <span class="dv">16</span>, ...)
  <span class="kw">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">col =</span> <span class="st">"tomato"</span>, <span class="dt">lwd =</span> <span class="dv">2</span>)
}
for(i in <span class="dv">1</span>:<span class="dv">2</span>) {
  <span class="kw">plotCompare</span>(<span class="kw">or2alpha</span>(result$y), paperResults[,<span class="dv">2</span>],
    <span class="dt">xlab =</span> <span class="st">"pulseR estimation"</span>,
    <span class="dt">ylab =</span> <span class="st">"paper"</span>,
    <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">"proportion methylated, batch "</span>, i))
  <span class="kw">plotCompare</span>(result$y, <span class="kw">lor</span>(paperResults[,<span class="dv">2</span>]),
    <span class="dt">xlab =</span> <span class="st">"pulseR estimation"</span>,
    <span class="dt">ylab =</span> <span class="st">"paper"</span>,
    <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">"log(odds) methylated, batch "</span>, i))
}</code></pre></div>
<p><img src="epitranscriptomics_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div id="confidence-intervals" class="section level3">
<h3 class="hasAnchor">
<a href="#confidence-intervals" class="anchor"></a>Confidence intervals</h3>
<p>In addition to the estimations, one usually is interested to have confidence intervals for making inference about the parameter values, for example, to compare between different conditions.</p>
<p>The function <code>ciGene</code> allows to estimate CI for the gene specific parameters. In this case, uncertainty from the normalisation factors is neglected and the profile of the likelihood is computed assuming the normalisation factors and the size parameter to be fixed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">opts$replicates &lt;-<span class="st"> </span><span class="dv">3</span>
cis &lt;-<span class="st"> </span><span class="kw"><a href="../reference/confint.html">ciGene</a></span>(<span class="dt">parName =</span> <span class="st">"y"</span>,
              <span class="dt">geneIndexes =</span> <span class="kw">seq_along</span>(result$y),
              <span class="dt">par =</span> result,
              <span class="dt">pd =</span> pd,
              <span class="dt">options =</span> opts,
              <span class="dt">interval =</span> <span class="kw">c</span>(-<span class="dv">5</span>, <span class="dv">5</span>),
              <span class="dt">confidence =</span> .<span class="dv">95</span>
)</code></pre></div>
<p>Here we gather the results into a data.frame and plot the confidence intervals and the estimated values. The genes are sorted by their m6A level:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
dat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(
  <span class="dt">y =</span> result$y,
  <span class="dt">mu =</span> result$mu,
  <span class="dt">y.min =</span> cis[,<span class="dv">1</span>],
  <span class="dt">y.max =</span> cis[,<span class="dv">2</span>]
)

dat$paperAvg &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(paperResults[,<span class="dv">2</span>:<span class="dv">3</span>])
dat$lor.paperAvg &lt;-<span class="st"> </span><span class="kw">lor</span>(dat$paperAvg)

<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>), <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">4</span>,<span class="dv">2</span>))
o &lt;-<span class="st"> </span><span class="kw">order</span>(dat$y)
i &lt;-<span class="st"> </span><span class="kw">seq_along</span>(dat$y)
<span class="kw">plot</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$y[o], <span class="dt">type =</span> <span class="st">"n"</span>,
  <span class="dt">xlab =</span> <span class="kw">expression</span>(<span class="kw">paste</span>(m^<span class="dv">6</span>, A, <span class="st">"  rank"</span>)),
  <span class="dt">ylab =</span> <span class="kw">expression</span>(<span class="kw">paste</span>(m^<span class="dv">6</span>, A, <span class="st">" log(odds)"</span>)),
  <span class="dt">cex.lab =</span> <span class="fl">1.2</span>
)
<span class="kw">lines</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$y.max[o], <span class="dt">col =</span> <span class="st">"dodgerblue"</span>)
<span class="kw">lines</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$y.min[o], <span class="dt">col =</span> <span class="st">"dodgerblue"</span>)
<span class="kw">points</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$y[o], <span class="dt">cex =</span> .<span class="dv">3</span>)

<span class="kw">plot</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> <span class="kw">or2alpha</span>(dat$y[o]), <span class="dt">type =</span> <span class="st">"n"</span>,
  <span class="dt">xlab =</span> <span class="kw">expression</span>(<span class="kw">paste</span>(m^<span class="dv">6</span>, A, <span class="st">"  rank"</span>)),
  <span class="dt">ylab =</span> <span class="kw">expression</span>(<span class="kw">paste</span>(m^<span class="dv">6</span>, A, <span class="st">"  proportion"</span>)),
  <span class="dt">cex.lab =</span> <span class="fl">1.2</span>
)
<span class="kw">lines</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> <span class="kw">or2alpha</span>(dat$y.max[o]), <span class="dt">col =</span> <span class="st">"dodgerblue"</span>)
<span class="kw">lines</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> <span class="kw">or2alpha</span>(dat$y.min[o]), <span class="dt">col =</span> <span class="st">"dodgerblue"</span>)
<span class="kw">points</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> <span class="kw">or2alpha</span>(dat$y[o]), <span class="dt">cex =</span> .<span class="dv">3</span>)</code></pre></div>
<p><img src="epitranscriptomics_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div id="on-bias-in-estimations" class="section level3">
<h3 class="hasAnchor">
<a href="#on-bias-in-estimations" class="anchor"></a>On bias in estimations</h3>
<p>As it was mentioned before, the normalisation may have high influence on the estimated values, which can shift the absolute values to higher or lower values, although the ranking is much less affected. To illustrate how estimations compare to each other, I add to the previous plot the average of the two estimations from the paper (red dots):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">4</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$y[o], <span class="dt">type =</span> <span class="st">"n"</span>,
  <span class="dt">xlab =</span> <span class="kw">expression</span>(<span class="kw">paste</span>(m^<span class="dv">6</span>, A, <span class="st">"  rank"</span>)),
  <span class="dt">ylab =</span> <span class="kw">expression</span>(<span class="kw">paste</span>(m^<span class="dv">6</span>, A, <span class="st">" log(odds)"</span>)),
  <span class="dt">cex.lab =</span> <span class="fl">1.2</span>
)
<span class="kw">lines</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$y.max[o], <span class="dt">col =</span> <span class="st">"dodgerblue"</span>)
<span class="kw">lines</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$y.min[o], <span class="dt">col =</span> <span class="st">"dodgerblue"</span>)
<span class="kw">points</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$y[o], <span class="dt">cex =</span> .<span class="dv">3</span>)
<span class="kw">points</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$lor.paperAvg[o], <span class="dt">cex =</span> .<span class="dv">3</span>, <span class="dt">col =</span> <span class="st">"tomato"</span>)
<span class="kw">legend</span>(<span class="st">"topright"</span>, <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">"pulseR"</span>, <span class="st">"paper"</span>), <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">"black"</span>, <span class="st">"tomato"</span>),
  <span class="dt">lty =</span> <span class="dv">1</span>)</code></pre></div>
<p><img src="epitranscriptomics_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.4.0 (2017-04-21)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 16.04.5 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS: /usr/lib/openblas-base/libblas.so.3</span>
<span class="co">#&gt; LAPACK: /usr/lib/libopenblasp-r0.2.18.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] pulseR_1.0.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_0.12.19       knitr_1.20         xml2_1.2.0        </span>
<span class="co">#&gt;  [4] magrittr_1.5       roxygen2_6.1.0     MASS_7.3-50       </span>
<span class="co">#&gt;  [7] R6_2.3.0           rlang_0.2.2        stringr_1.3.1     </span>
<span class="co">#&gt; [10] highr_0.7          tools_3.4.0        htmltools_0.3.6   </span>
<span class="co">#&gt; [13] commonmark_1.6     yaml_2.2.0         rprojroot_1.3-2   </span>
<span class="co">#&gt; [16] digest_0.6.18      assertthat_0.2.0   pkgdown_1.1.0.9000</span>
<span class="co">#&gt; [19] crayon_1.3.4       fs_1.2.6           memoise_1.1.0     </span>
<span class="co">#&gt; [22] evaluate_0.12      rmarkdown_1.10     stringi_1.2.4     </span>
<span class="co">#&gt; [25] compiler_3.4.0     desc_1.2.0         backports_1.1.2</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#analysis">Analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#get-and-prepare-the-data">Get and prepare the data</a></li>
      <li><a href="#which-parametrisation-to-use">Which parametrisation to use?</a></li>
      <li><a href="#model-definition">Model definition</a></li>
      <li><a href="#fitting-options">Fitting options</a></li>
      <li><a href="#results">Results</a></li>
      <li><a href="#confidence-intervals">Confidence intervals</a></li>
      <li><a href="#on-bias-in-estimations">On bias in estimations</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Alexey Uvarovskii.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
