<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The workflow • pulseR</title>
<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="The workflow">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pulseR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workflow.html">The workflow</a>
    </li>
    <li>
      <a href="../articles/epitranscriptomics.html">Analysis of RNA modifications</a>
    </li>
    <li>
      <a href="../articles/confidence.html">Profile likelihood</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dieterich-lab/pulseR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>The workflow</h1>
                        <h4 class="author">Uvarovskii Alexey</h4>
            
            <h4 class="date">2018-10-22</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dieterich-lab/pulseR/blob/master/vignettes/workflow.Rmd"><code>vignettes/workflow.Rmd</code></a></small>
      <div class="hidden name"><code>workflow.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>The workflow consists of defining the data object, setting fitting options and fitting itself:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(pulseR)
...
<span class="co"># define the data</span>
pd &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PulseData.html">PulseData</a></span>(...)
<span class="co"># set options</span>
opts &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setTolerance.html">setTolerance</a></span>(...)
opts &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setBoundaries.html">setBoundaries</a></span>(..., opts)
<span class="co"># define first parameter guess</span>
initPars &lt;-<span class="st"> </span><span class="kw"><a href="../reference/initParameters.html">initParameters</a></span>(...)
<span class="co"># fit the model</span>
fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/fit.html">fitModel</a></span>(pd, initPars, opts)</code></pre></div>
</div>
<div id="pulsedata-object" class="section level2">
<h2 class="hasAnchor">
<a href="#pulsedata-object" class="anchor"></a>PulseData object</h2>
<p>In the <code>pulseR</code> package, we keep all inial data in the <code>PulseData</code> structure (S3 class). When the information about read counts, conditions and the model is defined, one can create the object with a line as</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(pulseR)
<span class="kw">attach</span>(pulseRSpikeinsData)
pd &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PulseData.html">PulseData</a></span>(counts, conditions, formulas, formulaIndexes, spikeins)</code></pre></div>
<div id="counts" class="section level4">
<h4 class="hasAnchor">
<a href="#counts" class="anchor"></a>Counts</h4>
<p>The read counts must be delivered as an integer matrix. The rownames may describe the gene names and the column order corresponds to the sample order in the condition matrix (see below)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts[<span class="kw">c</span>(<span class="dv">1</span>:<span class="dv">5</span>, <span class="dv">50</span>:<span class="dv">55</span>), <span class="dv">1</span>:<span class="dv">4</span>]
##         total_fraction total_fraction total_fraction pull_down
## gene_1            1006            983           1005      1859
## gene_2            1984           1975           1992      3214
## gene_3            2914           2971           3049      4930
## gene_4            4106           4012           3839      7377
## gene_5            5083           5240           4915      8455
## gene_50          49745          50289          50003     80099
## gene_51          50713          52207          50233     94670
## gene_52          51829          51490          51292     87236
## gene_53          53077          52261          54400     96165
## gene_54          52626          52373          53856    102734
## gene_55          55412          55741          55124     89592</code></pre></div>
</div>
<div id="condition-matrix" class="section level4">
<h4 class="hasAnchor">
<a href="#condition-matrix" class="anchor"></a>Condition matrix</h4>
<p>The information about samples must be described in the condition matrix. It is obligatory that the row order corresponds to the column order in the count matrix. Time or other sample-specific variable are defined here.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">conditions
##                 fraction time
## sample_1  total_fraction    1
## sample_2  total_fraction    2
## sample_3  total_fraction    3
## sample_4       pull_down    1
## sample_5       pull_down    2
## sample_6       pull_down    3
## sample_7  total_fraction    1
## sample_8  total_fraction    2
## sample_9  total_fraction    3
## sample_10      pull_down    1
## sample_11      pull_down    2
## sample_12      pull_down    3</code></pre></div>
</div>
<div id="formulas" class="section level4">
<h4 class="hasAnchor">
<a href="#formulas" class="anchor"></a>Formulas</h4>
<p>The design of the experiment defines how different fractions evolve with the time. Let us consider the chase experiment, when the labelled molecules are being only degraded and the unlabelled molecules are synthesised and degraded:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">formulas &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MeanFormulas.html">MeanFormulas</a></span>(
  <span class="dt">total      =</span>  mu,
  <span class="dt">labelled   =</span>  mu *<span class="st"> </span><span class="kw">exp</span>(-d *<span class="st"> </span>time),
  <span class="dt">unlabelled =</span>  mu *<span class="st"> </span>(<span class="dv">1</span> -<span class="st"> </span><span class="kw">exp</span>(-d *<span class="st"> </span>time))
)</code></pre></div>
<p>Although the expression level in the form of mean read count is easier for perception, it can be better to model it at the logarithmic scale. In this case, the mean read counts is <code>exp(mu)</code>, and the degradation rate <code>d</code> and logarithm of mean read count <code>mu</code> have values of the same scale, i.e. unit changes in <code>d * time</code> or <code>mu</code> result in similar changes in the expected read number:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">formulas &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MeanFormulas.html">MeanFormulas</a></span>(
  <span class="dt">total      =</span>  <span class="kw">exp</span>(mu),
  <span class="dt">labelled   =</span>  <span class="kw">exp</span>(mu -<span class="st"> </span>d *<span class="st"> </span>time),
  <span class="dt">unlabelled =</span>  <span class="kw">exp</span>(mu) *<span class="st"> </span>(<span class="dv">1</span> -<span class="st"> </span><span class="kw">exp</span>(-d *<span class="st"> </span>time))
)</code></pre></div>
<p>In the mathematical form, it looks like <span class="math display">\[
\begin{align}
\text{[total]} &amp;= e^\mu\\
\text{[labelled]} &amp;= e^{\mu - d\cdot t}\\
\text{[unlabelled]} &amp;= e^{\mu}\left(1 - e^{-d\cdot t}\right)
\end{align}
\]</span></p>
<p><strong>In this vignette we use the logarithm of the mean read expression!</strong></p>
<p>It is important to note, that one should not forget to draw attention to to the correct scale of the parameter ranges, see below.</p>
</div>
<div id="fraction-content" class="section level4">
<h4 class="hasAnchor">
<a href="#fraction-content" class="anchor"></a>Fraction content</h4>
<p>In pulseR, it is possible to model contamination of the fractions. This definition is passed as <code>formulaIndexes</code> argument to the <code>PulseData</code> function. Here we define that the pull-down fraction consists of the labelled and unlabelled molecules, and the total fraction is degenerates to a simple formula, which we defined under the <code>total</code> list item in the <code>formulas</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">formulaIndexes &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">total_fraction =</span> <span class="st">"total"</span>,
                       <span class="dt">pull_down      =</span> <span class="kw">c</span>(<span class="st">"labelled"</span>, <span class="st">"unlabelled"</span>))</code></pre></div>
</div>
<div id="normalisation" class="section level4">
<h4 class="hasAnchor">
<a href="#normalisation" class="anchor"></a>Normalisation</h4>
<div id="using-spike-ins" class="section level5">
<h5 class="hasAnchor">
<a href="#using-spike-ins" class="anchor"></a>Using spike-ins</h5>
<p>Samples can be normalised using spike-ins (<strong>this vignette</strong>), which should be included in the count table together with other genes. One group of sample must be assigned as a reference, namely, the total fraction, which includes all the spike-ins type, labelled and unlabelled, in the original proportion.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">spikeins
## $refGroup
## [1] "total_fraction"
## 
## $spikeLists
## $spikeLists$total_fraction
## $spikeLists$total_fraction[[1]]
##  [1] "spikes 1"  "spikes 2"  "spikes 3"  "spikes 4"  "spikes 5" 
##  [6] "spikes 6"  "spikes 7"  "spikes 8"  "spikes 9"  "spikes 10"
## 
## 
## $spikeLists$pull_down
## $spikeLists$pull_down[[1]]
## [1] "spikes 1" "spikes 2" "spikes 3" "spikes 4" "spikes 5"
## 
## $spikeLists$pull_down[[2]]
## [1] "spikes 6"  "spikes 7"  "spikes 8"  "spikes 9"  "spikes 10"</code></pre></div>
<p>The fitting procedure uses as an input the <code>PulseData</code> object, which we can create having defined all the information about the experimental system:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pd &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PulseData.html">PulseData</a></span>(counts, conditions, formulas, formulaIndexes,
                <span class="dt">spikeins =</span> spikeins)</code></pre></div>
</div>
<div id="without-spike-ins" class="section level5">
<h5 class="hasAnchor">
<a href="#without-spike-ins" class="anchor"></a>Without spike-ins</h5>
<p>Alternatively, the relations between samples can be inferred during the fitting procedure (if no spike-ins counts are present in the data). There are two factors, which affects the expected read number in the model at a sample-wise level:</p>
<ul>
<li>
<strong>with-in group</strong>: Samples from the same group (e.g. “pull-down at 2 hr”) are normalised according to the sequencing depth using the same technique as in the DESeq package.</li>
<li>
<strong>between groups</strong> Normalisation factors, which define the relations between the groups and fractions are fitted together with other parameters.</li>
</ul>
<p>The way of sample grouping is defined in the <code>group</code> argument of the <code>PulseData</code> function. We assume that efficiency of the pull-down procedure is different between different time points (<code>time</code> column in the condition matrix), but the same for the samples from the same time point:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pd &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PulseData.html">PulseData</a></span>(counts, conditions, formulas, formulaIndexes,
                <span class="dt">groups =</span> ~<span class="st"> </span>fraction +<span class="st"> </span>time)</code></pre></div>
</div>
</div>
</div>
<div id="model-fitting" class="section level2">
<h2 class="hasAnchor">
<a href="#model-fitting" class="anchor"></a>Model fitting</h2>
<p>The parameters can be estimated using the <code>fitModel</code> procedure. It requires to provide the <code>PulseData</code> object, the initial guess for the parameter values (see below) and the fitting options.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/fit.html">fitModel</a></span>(<span class="dt">pulseData =</span> pd, <span class="dt">par =</span> initPars, <span class="dt">options =</span> opts)</code></pre></div>
<p>The function will iteratively fit the parameters from the formulas, overdispersion parameter for the negative binomial distribution and, if defined, the normalisation factors, until the stopping criteria is met.</p>
<div id="fitting-options" class="section level4">
<h4 class="hasAnchor">
<a href="#fitting-options" class="anchor"></a>Fitting options</h4>
<p>To fit the model, we use <em>L-BFGS-B</em> method implemented in the <code><a href="http://www.rdocumentation.org/packages/stats/topics/optim">stats::optim</a></code> function. It requires values for the lower and upper boundaries for the parameter values.</p>
<p>As it was mentioned earlier, we model the mean expression level at the logarithmic scale, which must be reflected in the boundaries as well. One may simply add <code>log</code> to the exprected read count range:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">opts &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setBoundaries.html">setBoundaries</a></span>(<span class="kw">list</span>(
  <span class="dt">mu =</span> <span class="kw">log</span>(<span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">1e6</span>)),
  <span class="dt">d  =</span> <span class="kw">c</span>(.<span class="dv">01</span>, <span class="dv">2</span>),
  <span class="dt">size =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">1e7</span>)
))</code></pre></div>
<p>The <code>fitModel</code> function performs fitting in iterations: it optimises gene-specific, shared, overdispersion and normalisation parameters at separate steps. Relative change between them is the stopping criteria:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">opts &lt;-<span class="st"> </span><span class="kw"><a href="../reference/setTolerance.html">setTolerance</a></span>(<span class="dt">params =</span> <span class="fl">1e-3</span>,  <span class="dt">options =</span> opts)</code></pre></div>
</div>
<div id="initial-values" class="section level4">
<h4 class="hasAnchor">
<a href="#initial-values" class="anchor"></a>Initial values</h4>
<p>The fitting routine needs an initial guess for the parameter values. The result of the fitting may depend on the model parametrisation and the initial guess.</p>
<p>The parameters, which are not specified in the par argument, in our case, both, the expression level and the degradation rate, will be assigned a random value within the boundaries defined in the fitting options list:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">initPars &lt;-<span class="st"> </span><span class="kw"><a href="../reference/initParameters.html">initParameters</a></span>(<span class="dt">par =</span> <span class="kw">list</span>(<span class="dt">size=</span><span class="fl">1e5</span>),
                           <span class="dt">geneParams =</span> <span class="kw">c</span>(<span class="st">"mu"</span>, <span class="st">"d"</span>),
                           <span class="dt">pulseData  =</span> pd,
                           <span class="dt">options    =</span> opts)</code></pre></div>
<p>However, since we use the total sample as a reference, we expect the read count in the “total_fraction” samples to be of close value to the optimal estimation of the expression level, i.e. <code>exp(mu)</code> in our parametrisation. We can help the fitting procedure by providing these values as a first guess (do not forget about the logarithm):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">initPars$mu &lt;-<span class="st"> </span><span class="kw">log</span>(pd$counts[,<span class="dv">1</span>] +<span class="st"> </span>.<span class="dv">5</span>)</code></pre></div>
</div>
<div id="result-exploration" class="section level4">
<h4 class="hasAnchor">
<a href="#result-exploration" class="anchor"></a>Result exploration</h4>
<p>The result of the <code>fitModel</code> function is just a list with the parameter values, which are ordered according to the rows in the read count table:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/fit.html">fitModel</a></span>(<span class="dt">pulseData =</span> pd, <span class="dt">par =</span> initPars, <span class="dt">options =</span> opts)
<span class="kw">str</span>(fit)
## List of 3
##  $ size: num 5471
##  $ mu  : num [1:500] 6.92 7.62 8.02 8.3 8.54 ...
##  $ d   : num [1:500] 0.161 0.272 0.237 0.119 0.208 ...</code></pre></div>
<p>We can use model prediction to see how good the model fits to the data (in the case of the simulated data all looks good):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/predictExpression.html">predictExpression</a></span>(fit, pd)
<span class="kw">plot</span>(pr$predictions, pd$counts, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">cex=</span>.<span class="dv">5</span>, <span class="dt">log=</span><span class="st">'xy'</span>,
     <span class="dt">xlab=</span><span class="st">"prediction"</span>, <span class="dt">ylab=</span><span class="st">"experiment"</span>)
<span class="kw">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">col =</span> <span class="st">"tomato"</span>)</code></pre></div>
<div class="figure">
<img src="workflow_files/figure-html/unnamed-chunk-18-1.png" alt="Model predictions vs. data" width="384"><p class="caption">
Model predictions vs. data
</p>
</div>
<p>The data were simulated from known values in <code>pulseRSpikeinsData$par</code> and we can compare fit to the true values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(par$d, fit$d, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">cex=</span><span class="dv">1</span>,
     <span class="dt">xlab=</span><span class="st">"true"</span>, <span class="dt">ylab=</span><span class="st">"fit"</span>, <span class="dt">main=</span><span class="st">"d, degradation rate"</span>)
<span class="kw">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">col =</span> <span class="st">"tomato"</span>)
<span class="kw">plot</span>(par$mu, <span class="kw">exp</span>(fit$mu), <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">cex=</span><span class="dv">1</span>, <span class="dt">log=</span><span class="st">'xy'</span>,
     <span class="dt">xlab=</span><span class="st">"true"</span>, <span class="dt">ylab=</span><span class="st">"fit"</span>, <span class="dt">main=</span><span class="st">"mu, expression level"</span>)
<span class="kw">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">col =</span> <span class="st">"tomato"</span>)</code></pre></div>
<div class="figure">
<img src="workflow_files/figure-html/unnamed-chunk-19-1.png" alt="Fitted vs. true values" width="768"><p class="caption">
Fitted vs. true values
</p>
</div>
<p>One can see, that there is a bias in the estimations: the normalisation affects the estimation by introducing a shift in the degradations rates. In this example, the normalisation and cross-contamination is derived on the basis of 5 labelled and 5 unlabelled spike-ins only.</p>
<p>For simulation the uniformly distributed parameters were used, which is also observed in the fitted values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">exp</span>(fit$mu), fit$d, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">cex=</span>.<span class="dv">5</span>,  <span class="dt">xlab=</span><span class="st">"expression"</span>, <span class="dt">ylab=</span><span class="st">"degradation"</span>)</code></pre></div>
<div class="figure">
<img src="workflow_files/figure-html/unnamed-chunk-20-1.png" alt="Parameter estimations" width="384"><p class="caption">
Parameter estimations
</p>
</div>
</div>
<div id="confidence-intervals" class="section level4">
<h4 class="hasAnchor">
<a href="#confidence-intervals" class="anchor"></a>Confidence intervals</h4>
<p>To generate the confidence intervals, one needs to provide the parameter name (e.g. “d”), indexes of the genes (in this example, only the first 150 genes), for which to compute the intervals and all the objects from the fitting procedure:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">geneSubset &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="dv">150</span>
cis &lt;-<span class="st"> </span><span class="kw"><a href="../reference/confint.html">ciGene</a></span>(<span class="st">"d"</span>, geneSubset, pd, fit, opts)

## Create a result table and plot the CI together with the estimations
dat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(
  <span class="dt">d  =</span> fit$d [geneSubset],
  <span class="dt">mu =</span> fit$mu[geneSubset],
  <span class="dt">d.min =</span> cis[,<span class="dv">1</span>],
  <span class="dt">d.max =</span> cis[,<span class="dv">2</span>]
)

o &lt;-<span class="st"> </span><span class="kw">order</span>(dat$d)
i &lt;-<span class="st"> </span><span class="kw">seq_along</span>(dat$d)
<span class="kw">plot</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$d[o], <span class="dt">type =</span> <span class="st">"n"</span>,
  <span class="dt">cex.lab =</span> <span class="fl">1.2</span>,
  <span class="dt">xlab =</span> <span class="st">"ranked by d"</span>,
  <span class="dt">ylab =</span> <span class="st">"d, 1/hr"</span>
)
<span class="kw">lines</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$d.max[o], <span class="dt">col =</span> <span class="st">"dodgerblue"</span>)
<span class="kw">lines</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$d.min[o], <span class="dt">col =</span> <span class="st">"dodgerblue"</span>)
<span class="kw">points</span>(<span class="dt">x =</span> i, <span class="dt">y =</span> dat$d[o], <span class="dt">cex =</span> .<span class="dv">3</span>)</code></pre></div>
<p><img src="workflow_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li>
<a href="#pulsedata-object">PulseData object</a><ul class="nav nav-pills nav-stacked">
<li><a href="#counts">Counts</a></li>
      <li><a href="#condition-matrix">Condition matrix</a></li>
      <li><a href="#formulas">Formulas</a></li>
      <li><a href="#fraction-content">Fraction content</a></li>
      <li><a href="#normalisation">Normalisation</a></li>
      </ul>
</li>
      <li>
<a href="#model-fitting">Model fitting</a><ul class="nav nav-pills nav-stacked">
<li><a href="#fitting-options">Fitting options</a></li>
      <li><a href="#initial-values">Initial values</a></li>
      <li><a href="#result-exploration">Result exploration</a></li>
      <li><a href="#confidence-intervals">Confidence intervals</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Alexey Uvarovskii.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
